src/crypto_helper/input/argon2.rs

```diff
@@ -0,0 +1,103 @@
+    crypto_helper::algorithm::{Argon2Action, Argon2HashAction, Argon2Input as Argon2InputData},
+};
+
+#[derive(PartialEq, Properties, Clone)]
+pub struct Argon2InputProps {
+    pub input: Argon2InputData,
+    pub argon2_input_setter: Callback<Argon2InputData>,
+}
+
+#[function_component(Argon2Input)]
+pub fn argon2_input(props: &Argon2InputProps) -> Html {
+    let data: Vec<u8> = props.input.data.clone();
+
+    let input_setter = props.argon2_input_setter.clone();
+    let action: Argon2Action = props.input.action.clone();
+    let setter = Callback::from(move |data| {
+        log::debug!("setting data: {data:?}");
+        input_setter.emit(Argon2InputData {
+            action: action.clone(),
+            data,
+        })
+    });
+
+    let data: Vec<u8> = props.input.data.clone();
+    let input_setter = props.argon2_input_setter.clone();
+    let action: Argon2Action = props.input.action.clone();
+
+    let salt_setter = Callback::from(move |salt| {
+        log::debug!("setting salt with data: {data:?}");
+        if let Argon2Action::Hash(hash_action) = action.clone() {
+            input_setter.emit(Argon2InputData {
+                action: Argon2Action::Hash(Argon2HashAction { salt, ..hash_action }),
+                data: data.clone(),
+            })
+        }
+    });
+
+    let input_setter = props.argon2_input_setter.clone();
+    let input = props.input.clone();
+    let on_hash_verify_switch = Callback::from(move |mode: bool| {
+        let Argon2InputData { data, .. } = input.clone();
+        input_setter.emit(Argon2InputData {
+            action: mode.into(),
+            data,
+        })
+    });
+    let hash_verify_switch_state = bool::from(&props.input.action);
+
+    let input_setter = props.argon2_input_setter.clone();
+    let input = props.input.clone();
+
+    let set_version = move |event: yew::html::onchange::Event| {
+        let html_element: HtmlInputElement = event.target_unchecked_into();
+        let input = input.set_version(html_element.value().as_str().try_into().unwrap());
+        input_setter.emit(input.clone());
+    };
+
+    let input_setter = props.argon2_input_setter.clone();
+    let input = props.input.clone();
+
+    let set_variant = move |event: yew::html::onchange::Event| {
+        let html_element: HtmlInputElement = event.target_unchecked_into();
+        let input = input.set_variant(html_element.value().as_str().try_into().unwrap());
+        input_setter.emit(input.clone());
+    };
+
+    let data: Vec<u8> = props.input.data.clone();
+    html! {
+        <div class={classes!("wide-input", "vertical")}>
+            <div class="horizontal">
+                <span class="total">{"hash"}</span>
+                <Switch id={"switch".to_string()} setter={on_hash_verify_switch} state={hash_verify_switch_state}/>
+                <span class="total">{"verify"}</span>
+            </div>
+            {build_byte_input(data, setter, Some(BytesFormat::Ascii), Some("argon2".into()))}
```

Comment:
add a hint for the user

```suggestion
            {build_byte_input(data, setter, Some(BytesFormat::Ascii), Some("password".into()))}
```